### PR TITLE
add auth_jwt_require directive

### PR DESCRIPTION
**Pull request by xynicole**
### Proposed changes

Added [auth_jwt_require](https://nginx.org/en/docs/http/ngx_http_auth_jwt_module.html#auth_jwt_require) directive

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-go-crossplane/blob/main/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-go-crossplane/blob/main/README.md))

### CODE REVIEW

**Review by nickchen — APPROVED** [review]

File: analyze.go [hunk 1]
```diff
@@ -2074,6 +2074,9 @@ var directives = map[string][]uint{
 	"auth_jwt_type": {
 		ngxHTTPMainConf | ngxHTTPSrvConf | ngxHTTPLocConf | ngxHTTPLmtConf | ngxConfTake1,
 	},
+	"auth_jwt_require": {
+		ngxHTTPMainConf | ngxHTTPSrvConf | ngxHTTPLocConf | ngxHTTPLmtConf | ngxConfTake1,
```
Comment (valyria257) [inline analyze.go:2078]: Can you check and see how many parameters you can use with `auth_jwt_require`? From the docs, it looks like more than one.

**Review by valyria257 — CHANGES_REQUESTED** [review]

File: analyze.go (hunk 1 above)
Comment (xynicole) [reply analyze.go]: you're right, it takes more than 1 args. so I changed to `ngxConf1More,`

**Review by valyria257 — APPROVED** [review]

File: analyze_test.go [hunk 2]
```diff
@@ -119,6 +119,50 @@ func TestAnalyze_auth_jwt(t *testing.T) {
 	}
 }
 
+func TestAnalyze_auth_jwt_require(t *testing.T) {
+	t.Parallel()
+	testcases := map[string]struct {
```
Comment (ornj) [inline analyze_test.go:124]: For completeness should there be a test to cover `auth_jwt_require $var error=403`?

**Review by ornj — APPROVED** [review]

File: analyze_test.go (hunk 2 above)
Comment (xynicole) [reply analyze_test.go:124]: done, added a test for the error code

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**